File: ruby/zenn_07_02.rb
# 次の連想配列(member)の中から
# 35歳以上の名前(name)の値だけを抜き取った
# 配列が返るような関数getNameOver35を作成して下さい。

members = [
  { name: "松井", age: 39, gender: "male" },
  { name: "今田", age: 34, gender: "female" },
  { name: "鈴木", age: 24, gender: "male" },
  { name: "山田", age: 56, gender: "male" },
  { name: "田中", age: 89, gender: "female" },
];

def getNameOver35(members)

  members.each do |member|
    if (member[:age] >= 35)
      puts member[:name]
    end
  end

end

getNameOver35(members)

File: php/zenn_07_02.php
<?php
# 次の連想配列(member)の中から
# 35歳以上の名前(name)の値だけを抜き取った配列が返るような
# 関数getNameOver35を作成して下さい。

$members = [
  [ "name" => "松井", "age" => 39, "sex" => "male" ],
  [ "name" => "今田", "age" => 34, "sex" => "female" ],
  [ "name" => "鈴木", "age" => 24, "sex" => "male" ],
  [ "name" => "山田", "age" => 56, "sex" => "male" ],
  [ "name" => "田中", "age" => 89, "sex" => "female" ]
];

function getNameOver35($members)
{
  foreach ($members as $member) {
    if ($member["age"] >= 35) {
      echo $member["name"] . "\n";
    }
  }
}

getNameOver35($members);
